application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,412 @@
+     * @param dateFormatterPattern the pattern used to parse the date from the database record
+     * @return the latest post date as a {@link ZonedDateTime} object, or null if the list is empty
+     */
+    private ZonedDateTime getLatestPostDateFromItems(List<Item> items,
+            String dateFormatterPattern) {
+        return items.stream()
+            .map(item -> getDateTimeFromItem(item, dateFormatterPattern))
+            .max(ZonedDateTime::compareTo)
+            .orElse(null);
+    }
+
+    /**
+     * Posts an RSS item to a text channel.
+     *
+     * @param textChannel the text channel to which the item will be posted
+     * @param rssItem the RSS item to post
+     * @param feedConfig the RSS feed configuration
+     */
+    private void postItem(TextChannel textChannel, Item rssItem, RSSFeed feedConfig) {
+        MessageEmbed embed = constructEmbedMessage(rssItem, feedConfig).build();
+        textChannel.sendMessageEmbeds(List.of(embed)).queue();
+    }
+
+    /**
+     * Updates the last posted date to the database for the specified RSS feed configuration.
+     * <p>
+     * This will insert a <b>new</b> entry to the database if the provided {@link RssFeedRecord} is
+     * null.
+     *
+     * @param feedConfig the RSS feed configuration
+     * @param rssFeedRecord the record representing the RSS feed. can be null if not found in the
+     *        database
+     * @param lastPostedDate the last posted date to be updated
+     */
+    private void updateLastDateToDatabase(RSSFeed feedConfig, @Nullable RssFeedRecord rssFeedRecord,
+            ZonedDateTime lastPostedDate) {
+        if (lastPostedDate == null) {
+            return;
+        }
+
+        DateTimeFormatter dateTimeFormatter =
+                DateTimeFormatter.ofPattern(feedConfig.dateFormatterPattern());
+        String lastDateStr = lastPostedDate.format(dateTimeFormatter);
+
+        if (rssFeedRecord == null) {
+            database.write(context -> context.newRecord(RSS_FEED)
+                .setUrl(feedConfig.url())
+                .setLastDate(lastDateStr)
+                .insert());
+            return;
+        }
+
+        database.write(context -> context.update(RSS_FEED)
+            .set(RSS_FEED.LAST_DATE, lastDateStr)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .executeAsync());
+    }
+
+    /**
+     * Attempts to get a {@link ZonedDateTime} from an {@link Item} with a provided string date time
+     * format.
+     * <p>
+     * If either of the function inputs are null, the oldest-possible {@link ZonedDateTime} will get
+     * returned instead.
+     *
+     * @param item The {@link Item} from which to extract the date.
+     * @param dateTimeFormat The format of the date time string.
+     * @return The computed {@link ZonedDateTime}
+     */
+    private static ZonedDateTime getDateTimeFromItem(Item item, String dateTimeFormat) {
```

Comment:
Can throw ZonedDateTime, specify that in the doc and throw clause.

Reply:
Still not fixed ?

Reply:
It's now fixed! Sorry about that :)